test/ui/item-not-fulfilled.test.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const assert = require('assert');
+const { By } = require('selenium-webdriver');
+
+const driverUtil = require('../../utils/driver');
+const { uiBaseUrl } = require('../../utils/base-urls');
+
+const customerService = require('../../services/customers');
+const orderService = require('../../services/orders');
+const productService = require('../../services/products');
+
+describe('Test that the UI responds properly when an item is not fulfilled', function () {
+  let orderId;
+  let customerId;
+  let url;
+  let driver;
+  before(async function() {
+    // build in an implicit 3 second wait so we don't have to wait for page to finish in our tests
+    driver = await driverUtil.buildChromeDriver({ implicit: 3000 });
+    const product = await productService.getProduct('2000');
+    const config = {
+      lines: [
+        {
+          variant_id: product.variant_id,
+          product_id: product.id,
+          quantity: 1,
+          fulfillable_quantity: 1,
+          fulfillment_status: 'null',
+          title: product.title,
+          variant_title: product.variant_title,
+          price: product.price,
+          sku: product.sku,
+          barcode: product.barcode,
+          vendor: '2000',
+        }
+      ]
+    };
+    const order = await orderService.createOrder(config);
+    orderId = order.id;
+    customerId = order.customer.id;
+    url = `${uiBaseUrl}/?orderId=${orderId}`;
+  });
+
+  after(async function() {
+    driver && driver.close();
+
+    if (orderId) {
+      await orderService.deleteOrder(orderId);
+    }
+
+    if (customerId) {
+      await customerService.thoroughDeleteCustomer(customerId);
```

Comment:
looking at the logic of thoroughDeleteCustomer, it seems possible we may need multiple retries to delete the customer, and there may be a failure in the end. So do you want to catch deleted = false here and do something, like logging (_or_ failing the suite, which could be too harsh given this is in test env)?

Reply:
`thoroughDeleteCustomer` is the retry method... it's not named the best, I was just getting mad at shopify for not letting me delete customers right away; it returns a 422 which basically is telling me: "We understand what you want, but we won't do it yet because there is some other linkage for the customer and the recently deleted order that hasn't cleared"
The caller of `thoroughDeleteCustomer` gets to deal with if the customer is deleted or not... if we need more retries, I can bump up the default or the individual test can bump up the number as well